File: python_roguelike/data/card_data.py
from dataclasses import dataclass, field
from typing import List
from .combat_action_data import CombatActionData
from .enums import CardType


@dataclass
class CardData:
    id: str = ""
    name: str = ""
    description: str = ""
    mana_cost: int = 0
    star_rating: int = 1
    type: CardType = CardType.Attack
    actions: List[CombatActionData] = field(default_factory=list)

File: src/Roguelike.Core/Core/Data/CardData.cs
using System.Collections.Generic;

namespace Roguelike.Data
{


    public class CardData
    {


        public string Id { get; set; }


        public string Name { get; set; }


        public string Description { get; set; }


        public int ManaCost { get; set; }


        public int StarRating { get; set; }


        public CardType Type { get; set; }


        public List<CombatActionData> Actions { get; set; } = new List<CombatActionData>();


        public CardData() { }
    }
}
